includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -404,7 +404,31 @@ public function admin_options() {
 		// Add notices to the WooCommerce Payments settings page.
 		do_action( 'woocommerce_woocommerce_payments_admin_notices' );
 
-		parent::admin_options();
+		if ( ! WC_Payments_Admin::is_grouped_settings_enabled() ) {
+			parent::admin_options();
+			return;
+		}
+
+		$this->output_payments_settings_screen();
+	}
+
+	/**
+	 * Generates markup for the settings screen.
+	 */
+	public function output_payments_settings_screen() {
+		// hiding the save button because the react container has its own.
+		global $hide_save_button;
+		$hide_save_button = true;
+
+		?>
+		<ul class="subsubsub">
```

Comment:
Sorry for not noticing this earlier - this seems good for a start, but I think in the end it might be best to use the proper WP hooks to add the sections, so that they're added with the correct styles and they're added also on the main payments page. In this case `woocommerce_get_sections_checkout`.
Right now, with the flag enabled, the "All payment methods" is not visible in the `/wp-admin/admin.php?page=wc-settings&tab=checkout` page.

Reply:
oh, right, I didn't realized those links were supposed to be displayed too in the main payments page! :facepalm: No worries, I'll update the hook.

Reply:
I'm also good with getting this merged so that it can unblock subsequent work 👍  we can always revisit the hook later, since all of this is behind a feature flag

Reply:
Added the sections with https://github.com/Automattic/woocommerce-payments/pull/1580